_posts/2019-03-14-2019-kafka-connector.md

```diff
@@ -0,0 +1,194 @@
+
+Kafka is stream-processing platform and at a high level the important concepts are Producers, Consumers and Brokers which communicate with each-other via massages attached to Topics. The way it works is that the Producer is sending messages on specific Topics to the Broker, then the Consumers poll all the topics of interest from the Brokers. The approach is fitting for distributed systems, because it decouples the communication between your services.
+
+You can find out more by heading to the [Apache Kafka Documentation](https://kafka.apache.org/documentation/).
+
+## Kafka Connector
+
+Kafka Connector plugin is the solution to your problem when it comes to integrating OpenFaaS as a component of a bigger distributed system which is using Kafka to make communication between its parts. You just need to apply annotation with the topic you would like to subscribe your function to and then run the connector with your Kafka configuration.
+
+Here is a diagram which shows how the component works:
+
+![Form Options](/images/kafka-connector/kafka-connector-detailed-overview.png)
+
+## Pre-requisites
+
+First and most important thing is [running OpenFaaS instance](https://docs.openfaas.com/deployment/kubernetes/) with default configuration on Kubernetes, for our case [minikube deployment](https://kubernetes.io/docs/setup/minikube/) is more than enough.
+
+The [faas-cli](https://docs.openfaas.com/cli/install/) binary along with populated `OPENFAAS_URL` environmental variable pointing at your gateway.
+
+## Clone the Kafka Connector
+
+Pull the `kafka-connector` code from github:
+
+```bash
+$ git clone https://github.com/openfaas-incubator/kafka-connector
+```
+
+Inside `kafka-connector` folder navigate to `yaml/kubernetes`:
+
+```bash
+$ cd kafka-connector/yaml/kubernetes
+```
+
+Here we can see the Kafka Zookeeper and Broker which we will be using to send payloads to our functions.
+
+## Deploy the components with configuration
+
+In order to deploy the Zookeeper and Broker apply the kubernetes files while inside the kubernetes folder from above:
+
+```bash
+$ kubectl apply -f kafka-broker-dep.yml,kafka-broker-svc.yml,zookeeper-dep.yaml,zookeeper-svc.yaml
+```
+
+It is important to note that those files deploy the Zookeeper and Broker in the same namespace as the OpenFaaS Gateway.
+
+We will use charts to deploy the Kafka Connector. If you don't have the OpenFaaS charts repository, then add it with this command:
+
+```bash
+$ helm repo add openfaas https://openfaas.github.io/faas-netes/
+```
+
+Install Kafka Connector with default values:
+
+```bash
+$ helm upgrade kafka-connector openfaas/kafka-connector \
+    --install \
+    --namespace openfaas \
+    --set topics="faas-request\," \
+    --set broker_host="kafka" \
+    --set print_response_body="true"
```

Comment:
`--set print_response_body="true"` wont be applied - it doesn't exist in `values.yml`

Reply:
Added [PR](https://github.com/openfaas/faas-netes/pull/408)